rotkehlchen/chain/ethereum/modules/curve/curve_cache.py

```diff
@@ -241,172 +262,196 @@ def save_curve_data_to_cache(
+            )
+        except InputError:
+            # InputError can happen if name for the current pool/gauge is already known.
+            # It is not a bug (e.g. because user could have added the name before).
+            log.debug(f'Names for pool {pool.pool_name}({pool.pool_address}) alrady exist in the address book.')  # noqa: E501
+
         globaldb_set_general_cache_values(
             write_cursor=write_cursor,
-            key_parts=[GeneralCacheType.CURVE_POOL_ADDRESS, lp_token_address],
-            values=[pool_address],
+            key_parts=[GeneralCacheType.CURVE_LP_TOKENS],
+            values=[pool.lp_token_address],  # keys of pools_mapping are lp tokens
         )
-        if pool_address in gauges:
+        globaldb_set_general_cache_values(
+            write_cursor=write_cursor,
+            key_parts=[GeneralCacheType.CURVE_POOL_ADDRESS, pool.lp_token_address],
+            values=[pool.pool_address],
+        )
+        for i, coin in enumerate(pool.coins):
             globaldb_set_general_cache_values(
                 write_cursor=write_cursor,
-                key_parts=[GeneralCacheType.CURVE_GAUGE_ADDRESS, pool_address],
-                values=[gauges[pool_address]],
+                key_parts=[GeneralCacheType.CURVE_POOL_TOKENS, pool.pool_address, str(i)],
+                values=[coin],
             )
-        for index, coin in enumerate(coins):
+        if pool.gauge_address is not None:
             globaldb_set_general_cache_values(
                 write_cursor=write_cursor,
-                key_parts=[GeneralCacheType.CURVE_POOL_TOKENS, pool_address, str(index)],
-                values=[coin],
+                key_parts=[GeneralCacheType.CURVE_GAUGE_ADDRESS, pool.pool_address],
+                values=[pool.gauge_address],
             )
 
 
-def query_curve_registry_pools(
-        ethereum: 'EthereumInquirer',
-        registry_contract: EvmContract,
-) -> CURVE_POOLS_MAPPING_TYPE:
-    """Query pools from curve registry and return their mappings
-    May raise:
-    - RemoteError if failed to query etherscan or node
-    """
-    registry_pool_count = registry_contract.call(
+def query_curve_data_from_api() -> list[CurvePoolData]:
+    log.debug('Querying curve pools from curve api')
+    all_api_pools = []
+    for api_url in CURVE_API_URLS:
+        try:
+            raw_response = requests.get(api_url, timeout=DEFAULT_TIMEOUT_TUPLE)
+            if raw_response.status_code != 200:
+                raise RemoteError(f'Curve api returned unexpected status code {raw_response.status_code}')  # noqa: E501
+            response_json = raw_response.json()
+        except RequestException as e:
+            raise RemoteError(f'Could not query curve api due to: {str(e)}') from e
+
+        if response_json['success'] is False:
+            raise RemoteError('Curve api returned success=False. Bailing.')
+
+        try:
+            all_api_pools.extend(response_json['data']['poolData'])
```

Comment:
do you need to do this though? I mean creating a new variable for validation?
You can easily put the for loop in a try/except and return an empty list if keyError is raised.

Reply:
Not sure that I got the question. Here I get list of pools, combine all pools from all endpoints in a single list and then iterate them all at once. What do you suggest to change here?